test/subscription_test.go

```diff
@@ -0,0 +1,174 @@
+package test
+
+import (
+	"context"
+	"net/http"
+	"testing"
+	"time"
+
+	advancedbilling "github.com/maxio-com/ab-golang-sdk"
+	"github.com/maxio-com/ab-golang-sdk/models"
+)
+
+func (s *APISuite) TestSubscriptionCreate() {
+	ctx := context.Background()
+
+	customer := s.createCustomer(ctx)
+	productFamily := s.createProductFamily(ctx)
+	product := s.createProduct(ctx, *productFamily.Id)
+	coupon := s.createCoupon(ctx, *productFamily.Id)
+	component := s.createMeteredComponent(ctx, *productFamily.Id)
+
+	cases := []struct {
+		name         string
+		client       advancedbilling.ClientInterface
+		subscription models.CreateSubscription
+		assert       func(*testing.T, models.ApiResponse[models.SubscriptionResponse], models.CreateSubscription, error)
+	}{
+		{
+			name:   "valid",
+			client: s.client,
+			subscription: s.newSubscription(
+				customer,
+				product,
+				*coupon.Code,
+				[]models.CreateSubscriptionComponent{
+					{
+						ComponentId: interfacePtr(*component.Id),
+						Enabled:     boolPtr(true),
+						UnitBalance: intPtr(1),
+					},
+				},
+			),
+			assert: func(t *testing.T, resp models.ApiResponse[models.SubscriptionResponse], subscription models.CreateSubscription, err error) {
+				s.NoError(err)
+				s.Equal(http.StatusCreated, resp.Response.StatusCode)
+
+				createdSubscription := resp.Data.Subscription
+				s.Equal(models.SubscriptionState_AWAITINGSIGNUP, *createdSubscription.State)
+
+				s.Equal(subscription.ProductId, createdSubscription.Product.Id)
+				s.Equal(subscription.PaymentCollectionMethod, createdSubscription.PaymentCollectionMethod)
+				s.Equal(subscription.CustomerId, createdSubscription.Customer.Id)
+				s.Equal(subscription.Currency, createdSubscription.Currency)
+				s.Equal(subscription.CouponCode, createdSubscription.CouponCode.Value())
+				s.Equal(subscription.CustomerId, createdSubscription.Customer.Id)
+				s.Equal(subscription.ProductId, createdSubscription.Product.Id)
+
+				listResp, err := s.client.SubscriptionComponentsController().ListSubscriptionComponents(
```

Comment:
Is there some possibility in the SDK to provide only required parameters to avoid long list of nils?
Builder or some other pattern?

Reply:
Not in this version of sdk. We could use those parameters as options but that would require writing wrapper around sdk methods